stubs/wurlitzer/wurlitzer/__init__.pyi

```diff
@@ -0,0 +1,41 @@
+import contextlib
+import io
+from _typeshed import Incomplete
+from collections.abc import Iterator
+from contextlib import contextmanager
+from types import TracebackType
+from typing import Final, Literal, TextIO
+from typing_extensions import TypeAlias
+
+STDOUT: Final = 2
+PIPE: Final = 3
+_STDOUT: TypeAlias = Literal[2]
+_PIPE: TypeAlias = Literal[3]
+
+class Wurlitzer:
+    flush_interval: float = 0.2
+
+    def __init__(
+        self,
+        stdout: TextIO | io.IOBase | None = None,
+        stderr: _STDOUT | TextIO | io.IOBase | None = None,
+        encoding: str = ...,
+        bufsize: int | None = ...,
+    ) -> None: ...
+    def __enter__(self): ...
+    def __exit__(
+        self, exc_type: type[BaseException] | None, exc_value: BaseException | None, traceback: TracebackType | None
+    ) -> None: ...
+
+def dup2(a: int, b: int, timeout: int = 3) -> int: ...
+def sys_pipes(encoding: str = ..., bufsize: int | None = None) -> contextlib._GeneratorContextManager[None]: ...
+@contextmanager
+def pipes(
+    stdout: _PIPE | TextIO = 3, stderr: _STDOUT | _PIPE | TextIO = 3, encoding: str = ..., bufsize: int | None = None
+) -> Iterator[tuple[TextIO | io.BytesIO | io.StringIO, TextIO | io.BytesIO | io.StringIO | None]]: ...
+def sys_pipes_forever(encoding: str = ..., bufsize: int | None = None): ...
+def stop_sys_pipes() -> None: ...
+def load_ipython_extension(ip: Incomplete) -> None: ...  # IPython.core.interactiveshell.InteractiveShell
+def unload_ipython_extension(ip: Incomplete) -> None: ...  # IPython.core.interactiveshell.InteractiveShell
```

Comment:
Done in 6bc4a65b5.

Those functions are not documented and are not int the `__all__` export, so I'm not sure if they should be included in the stubs. They are not private, so I added them for completeness.

Reply:
Hmm. I'll leave it up to you. Our CI won't complain if they're deleted from the stubs, since they're not included in `__all__`. But some people could possibly find them useful, and having them in the stubs won't be a big maintenance burden for us. Overall I have no strong opinion either way :)